src/Symfony/Component/Form/Extension/Core/EventListener/ResizeFormListener.php

```diff
@@ -125,7 +128,7 @@ public function preSubmit(FormEvent $event)
                 if (!$form->has($name)) {
                     $form->add($name, $this->type, array_replace(array(
                         'property_path' => '['.$name.']',
-                    ), $this->options));
+                    ), $this->getOptionsForEntry($value)));
```

Comment:
`prototype_data` accepts data in the model format. So you could actually pass an object, which would be converted to the right view format by the prototype field.

The problem we have here is that the field which is responsible for converting the view data back to the model data has not been created yet. At the same time, we cannot create the field yet since we still need to determine the options for the field creation by inspecting the model data. We have a circular dependency.